### ISSUE TITLE
event_type determination should be done bitwise

### ISSUE THREAD

**Issue by moderski** [cta-sst-1m/digicampipe#244]
Muon flagging on camera server is going to be implemented. It is assumed that the algorithm will set new bits in camera_event_type field:
```
#define EVTYPE_CSP_MUON         (0x10000)       /* camera server detected muon*/
#define EVTYPE_CSP_HILLAS       (0x20000)       /* camera server computed Hillas parametrs */
```
additionally to already existing set of bits:
```
#define EVTYPE_DIGICAM_ALG0     (0x1)           /* algorithm 0 trigger - PATCH7 */
#define EVTYPE_DIGICAM_ALG1     (0x2)           /* algorithm 1 trigger - PATCH19 */
#define EVTYPE_DIGICAM_ALG2     (0x4)           /* algorithm 2 trigger - MUON */
#define EVTYPE_DIGICAM_INTRNL   (0x8)           /* internal or external  */
#define EVTYPE_DIGICAM_EXTMSTR  (0x10)          /* unused (0) / external (on master only)  */
#define EVTYPE_DIGICAM_BIT5     (0x20)          /* unused (0)  */
#define EVTYPE_DIGICAM_BIT6     (0x40)          /* unused (0)  */
#define EVTYPE_DIGICAM_CONT     (0x80)          /* continuous readout marker  */
```

Since many event_type assessments in digicampipe are done numberwise `if event_type == 1:` they should be changed to bitwise `if event_type & 0x1:`

**Comment by calispac** [comment]
Ok good point ! After some offline discussion we thought we could maybe use the `in` statement in python.

e.g. `if event_type.CSP_MUON in event_type`. To do this one would need to create a class which has a function `__contains__()` defined.

**Comment by dneise** [comment]
I looked further and found a nice solution for this in the Python stdlib module `enum`. The class doing this is called `IntFlag` 

So I propose this class:
```{python}
class CameraEventType(enum.IntFlag):
    '''
    I removed the EVTTYPE, since this is clear from the class name
    I removed DIGICAM since this is clear from the project.
    '''
    ALG0 = 0x1
    ALG1 = 0x2
    ALG2 = 0x4
    INTRNL = 0x8
    EXTMSTR = 0x10
    BIT5 = 0x20
    BIT6 = 0x40
    CONT = 0x80
    CSP_MUON = 0x10000
    CSP_HILLAS = 0x20000
```

It can be used like this: ...

**Comment by dneise** [comment]
Assume in the old code you had an event, which was triggered by ALG0 **and** its hillas parameters were calculated **and** it was flagged as a muon. Then in the previous version the value of event_type would have been:
    
    event_type = 0x1 + 0x10000 + 0x20000 = 0x30001

With this class one can do:
```python
In [21]: event_type = CameraEventType(0x30001)

In [22]: event_type
Out[22]: <CameraEventType.CSP_HILLAS|CSP_MUON|ALG0: 196609>

In [23]: CameraEventType.CSP_MUON in event_type
Out[23]: True

In [24]: CameraEventType.ALG1 in event_type
Out[24]: False
```

**Comment by dneise** [comment]
Instead of `enum.IntFlag` we can also use `enum-Flag`, this class even throws an error when we try to create a `CameraEventType` with a value, which makes no sense. So for example:

    event_type = CameraEventType(0x02000)

makes no sense since this bit is not defined and if CameraEventType inherits from `enum.Flag` we get an exception when we try to assign non-sensical values here. Nice, hu?

**Comment by dneise** [comment]
fixed in #245
